### PR TITLE
Implement syncInfo subscription & GlobalHub

### PR DESCRIPTION
**Pull request by dwalintukan**
moved syncInfo and wallet-related balance calls to an invisible component to handle this.

### CODE REVIEW

**Comment by liviavinci** [comment]
LGTM, leaving it to other reviewers to double check the biz logic since i'm still catching up

File: src/scenes/App/globalHub.jsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  render() {
+    return null;
+  }
+
+  subscribeSyncInfo() {
+    const { client, onSyncInfo } = this.props;
+
+    console.log('Subscribe: onSyncInfo');
```
Comment (liviavinci) [inline src/scenes/App/globalHub.jsx]: Remove

File: src/scenes/App/globalHub.jsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    // Get unspent outputs and bot balances for all the wallet addresses
+    this.updateBalances(walletAddrs);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { initSyncing, syncBlockNum } = this.props;
+
+    // Disable the syncInfo polling since we will get new syncInfo from the subscription
+    if (initSyncing && !nextProps.initSyncing) {
+      console.debug('Disable syncInfo polling');
```
Comment (liviavinci) [inline src/scenes/App/globalHub.jsx]: Remove

File: src/scenes/App/globalHub.jsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      initSyncing,
+      listUnspent,
+      getBotBalance,
+    } = this.props;
+
+    listUnspent();
+
+    if (!_.isEmpty(walletAddresses)) {
+      _.each(walletAddresses, (address) => {
+        getBotBalance(address.address, address.address);
```
Comment (liviavinci) [inline src/scenes/App/globalHub.jsx]: address.address is kinda weird, maybe wallet.address or something?

File: src/scenes/App/globalHub.jsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+  subscribeSyncInfo() {
+    const { client, onSyncInfo } = this.props;
+
+    console.log('Subscribe: onSyncInfo');
+    client.subscribe({
+      query: getSubscription(channels.ON_SYNC_INFO),
+    }).subscribe({
+      next(data) {
+        console.debug('onSyncInfo', data.data.onSyncInfo);
```
Comment (liviavinci) [inline src/scenes/App/globalHub.jsx]: remove

Comment (liviavinci) [inline src/scenes/App/globalHub.jsx]: remove

File: src/scenes/App/globalHub.jsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        onSyncInfo({ error: err.message });
+      },
+    });
+  }
+
+  pollSyncInfo() {
+    const { getSyncInfo } = this.props;
+
+    getSyncInfo();
+    console.debug('Starting syncInfo polling');
```
Comment (liviavinci) [inline src/scenes/App/globalHub.jsx]: remove

File: src/scenes/App/globalHub.jsx (hunk 1 above)
Comment (dwalintukan) [reply src/scenes/App/globalHub.jsx]: removed

File: src/scenes/App/globalHub.jsx (hunk 3 above)
Comment (dwalintukan) [reply src/scenes/App/globalHub.jsx]: fixed

File: src/scenes/App/globalHub.jsx (hunk 4 above)
Comment (dwalintukan) [reply src/scenes/App/globalHub.jsx]: removed

Comment (dwalintukan) [reply src/scenes/App/globalHub.jsx]: removed

File: src/scenes/App/globalHub.jsx (hunk 5 above)
Comment (dwalintukan) [reply src/scenes/App/globalHub.jsx]: removed

**Pull request merged**